src/vm/assembly.cpp

```diff
@@ -331,6 +337,27 @@ void Assembly::StartUnload()
         ProfilerCallAssemblyUnloadStarted(this);
     }
 #endif
+
+#ifdef FEATURE_COMINTEROP
+    // Release tlb files eagerly
+    if (g_fProcessDetach == FALSE)
+    {
+        DefaultCatchFilterParam param;
+        param.pv = COMPLUS_EXCEPTION_EXECUTE_HANDLER;
+        PAL_TRY(Assembly *, pThis, this)
```

Comment:
This looks like a hack to workaround a memory corruption (it can very well be the memory corruption from the race).

We should not be adding hacks like this to .NET Core. If there is a memory corruption, just let the process crash.